fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -572,8 +586,12 @@ def execute(command, hide_output)
   end
 
   class ItunesTransporter
-    # Matches a line in the provider table: "12  Initech Systems Inc     LG89CQY559"
-    PROVIDER_REGEX = /^\d+\s{2,}.+\s{2,}[^\s]+$/
+    # Matches a line in the iTMSTransporter provider table: "12  Initech Systems Inc     LG89CQY559"
+    ITMS_PROVIDER_REGEX = /^\d+\s{2,}.+\s{2,}[^\s]+$/
+    # Matches a line in the altool provider table's separator
+    # ------------------- ----------------- ------------------------------------ ----------
+    # Initech Systems Inc LG89CQY559        xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx LG89CQY559
+    ALTOOL_SEPARATOR_PROVIDER_REGEX = /^\-{1,}\s{1}\-{1,}\s{1}\-{1,}\s{1}\-{1,}\s+$/
```

Comment:
Couldn't we use `--output-format json` when invoking `altool --list-providers` and parse that as JSON instead of text? Would feel more reliable than this if it's possible?

Reply:
> `--output-format json`

wow, I did not know such great option. I lost time by making so complex regex parser ...
Thank you!!!!

Reply:
Yeah only discovered it 10mn ago by accident too when I wondered if `altool` didn't offer such option and looked at its help page for it. I wonder if it couldn't even help with the regular upload step as well (eg to parse `*** Error` logs etc, in case that json output present those in an easier-to-parse format after all)?

That being said, we could also merge the PR as is to avoid blocking, and open a separate subsequent PR to improve on that if you prefer not spending too much time on this now.

Reply:
it can be done easily by https://github.com/fastlane/fastlane/pull/20631/commits/4f4d3eefded04f86147bfa044a9b62fd542d7420, and also let me check how can I improve it in Error case.

Reply:
> That being said, we could also merge the PR as is to avoid blocking, and open a separate subsequent PR to improve on that if you prefer not spending too much time on this now.

I decided to do it later PR because it makes some other additional changes 🙇 since Xcode 14 is now released, it may be better to merge it as possible

Reply:
Good call, works for me 👍 

Updated [checklist above](https://github.com/fastlane/fastlane/pull/20631#pullrequestreview-1107282582) :sigh